external/pdal_wrench/translate.cpp

```diff
@@ -0,0 +1,172 @@
+/*****************************************************************************
+ *   Copyright (c) 2023, Lutra Consulting Ltd. and Hobu, Inc.                *
+ *                                                                           *
+ *   All rights reserved.                                                    *
+ *                                                                           *
+ *   This program is free software; you can redistribute it and/or modify    *
+ *   it under the terms of the GNU General Public License as published by    *
+ *   the Free Software Foundation; either version 3 of the License, or       *
+ *   (at your option) any later version.                                     *
+ *                                                                           *
+ ****************************************************************************/
+
+#include <iostream>
+#include <filesystem>
+#include <thread>
+
+#include <pdal/PipelineManager.hpp>
+#include <pdal/Stage.hpp>
+#include <pdal/util/ProgramArgs.hpp>
+#include <pdal/pdal_types.hpp>
+#include <pdal/Polygon.hpp>
+
+#include <gdal/gdal_utils.h>
+
+#include "utils.hpp"
+#include "alg.hpp"
+#include "vpc.hpp"
+
+using namespace pdal;
+
+namespace fs = std::filesystem;
+
+
+void Translate::addArgs()
+{
+    argOutput = &programArgs.add("output,o", "Output point cloud file", outputFile);
+    argOutputFormat = &programArgs.add("output-format", "Output format (las/laz/copc)", outputFormat);
+    programArgs.add("assign-crs", "Assigns CRS to data (no reprojection)", assignCrs);
+    programArgs.add("transform-crs", "Transforms (reprojects) data to anoter CRS", transformCrs);
+}
+
+bool Translate::checkArgs()
+{
+    if (!argOutput->set())
+    {
+        std::cerr << "missing output" << std::endl;
+        return false;
+    }
+
+    // TODO: or use the same format as the reader?
+    if (argOutputFormat->set())
+    {
+        if (outputFormat != "las" && outputFormat != "laz")
+        {
+            std::cerr << "unknown output format: " << outputFormat << std::endl;
+            return false;
+        }
+    }
+    else
+        outputFormat = "las";  // uncompressed by default
+
+    return true;
+}
+
+
+static std::unique_ptr<PipelineManager> pipeline(ParallelJobInfo *tile, std::string assignCrs, std::string transformCrs)
+{
+    std::unique_ptr<PipelineManager> manager( new PipelineManager );
+
+    Options reader_opts;
+    if (!assignCrs.empty())
+        reader_opts.add(pdal::Option("override_srs", assignCrs));
+
+    Stage& r = manager->makeReader( tile->inputFilenames[0], "", reader_opts);
+
+    // optional reprojection
+    Stage* reproject = nullptr;
+    if (!transformCrs.empty())
+    {
+        Options transform_opts;
+        transform_opts.add(pdal::Option("out_srs", transformCrs));
```

Comment:
This should really take a proj pipeline string instead of the destination crs -- that would mean we could ensure that the transformation correctly respects user transformation settings and matches the behaviour of raster/vector layers in processing

Reply:
You're right, it is in the queue to be added (using PDAL's `filters.projpipeline`)

Reply:
Addressed in https://github.com/PDAL/wrench/commit/a5b25d7152011b51170f7da80e781f78c4605948